lib/auth/auth.go

```diff
@@ -4323,6 +4352,45 @@ func (a *Server) UpsertWindowsDesktop(ctx context.Context, desktop types.Windows
 	return nil
 }
 
+// desktopsLimitExceeded checks if number of non-AD desktops exceeds limit for OSS distribution. Returns always false for Enterprise.
+// If there are more non-AD desktops than allowed it also adds cluster alert to notify user.
+func (a *Server) desktopsLimitExceeded(ctx context.Context) (bool, error) {
```

Comment:
Should we separate the check from the alert upsert? The name of this method sounds like it just checks the limit.

Reply:
Yes, I've separated it.